### PR TITLE
EDITOR

### PR DESCRIPTION
**Pull request by SAaradhya**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,66 @@
 # Wall-Art-Bot
 Open Projects 2021
+
+## Abstract
+The main aim of this project is to draw a desired sketch on a wall.
```
Comment (theobscuredev) [inline README.md:5]: Add a new line between this text and the following Image.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,66 @@
...
+![](Image%20and%20Videos/Mechanical%20Design%20Images/Multiple%20view.png)
```
Comment (theobscuredev) [inline README.md]: Center align all the images using the `<p>` paragraph tag and `align` attribute. For eg:
`<p align="center">`
  `LED Matrix`
  `<i>LED Matrix</i>`
`</p>`

Also resize the images by assigning the dimensions.

File: README.md [hunk 3]
```diff
@@ -1,2 +1,66 @@
...
+## Components and cost
+* 2 x Nema 11/17 Stepper Motor       - Rs.1000                                     
+* 1x MG90S Servo Motor               - Rs.250                                     
+* 1x Arduino Mega                    - Rs.1000                                     
+* 1x L293D Motor Drive Shield        - Rs.150                                    
+* 1x 28B28BYJ-48 DC 5V Stepper Motor - Rs.150
+  + ULN2003 Driver Test Module Board
+* 2 X Command Large Plastic Hooks for walls -Rs. 140
+* Rope - DYNAMICA XBO       	     - Rs.350
+* Total- Rs.3040
```
Comment (theobscuredev) [inline README.md]: Tabulate the Cost Structure table. Refer this for the markdown syntax: https://raw.githubusercontent.com/marsiitr/Rube-Goldberg-Machine/master/README.md

Also cost structure should be included after the Mechanical & Electronic Aspects. Refer the documentation for the same.

File: README.md [hunk 4]
```diff
@@ -1,2 +1,66 @@
 # Wall-Art-Bot
 Open Projects 2021
+
+## Abstract
+The main aim of this project is to draw a desired sketch on a wall.
+![](Image%20and%20Videos/Mechanical%20Design%20Images/Multiple%20view.png)
+
+## Motivation
+We always want to draw pictures from what we see scrolling on the web. But it's always difficult to convince the professional painter to draw the exact same picture on our walls. This project helps users to draw the exact same sketch that they have uploaded.
```
Comment (theobscuredev) [inline README.md]: Justify the entire text using the `<p>` paragraph tag and `justify` attribute. Make sure you justify all the paragraphs separately. For eg:
`<p align="justify">`
  `<h2>Abstract</h2>`
  `<p>A Rube Goldberg Machine is .........................</p>`
`</p>`

Refer this README for the same: https://raw.githubusercontent.com/marsiitr/Rube-Goldberg-Machine/master/README.md

File: README.md [hunk 5]
```diff
@@ -1,2 +1,66 @@
...
+* 1x L293D Motor Drive Shield        - Rs.150                                    
+* 1x 28B28BYJ-48 DC 5V Stepper Motor - Rs.150
+  + ULN2003 Driver Test Module Board
+* 2 X Command Large Plastic Hooks for walls -Rs. 140
+* Rope - DYNAMICA XBO       	     - Rs.350
+* Total- Rs.3040				     
+## Workflow
+   ![](Report%20and%20Poster/workflow.jpg)
+## Mechanical Aspect of the Design
+   #### 1.Pulley Movement
```
Comment (theobscuredev) [inline README.md:46]: Insert a space for aesthetic purposes only. For eg:
1. Pulley Movement

File: README.md [hunk 6]
```diff
@@ -1,2 +1,66 @@
...
+   Used Stepper motors to rotate pulleys and helps the pen to reach desired coordinate.
+   ![](Image%20and%20Videos/Mechanical%20Design%20Images/Pulley%20movement.png)
+   #### 2.Pen Module
```
Comment (theobscuredev) [inline README.md:53]: Insert a space for aesthetic purposes only.

File: README.md [hunk 7]
```diff
@@ -1,2 +1,66 @@
...
+   The pen module which is a disc contains 3 pen holders and the disc rotates using a stepper motor and then there are ball bearings that ease the rotation of the disc.
+   
+   ![](Image%20and%20Videos/Mechanical%20Design%20Images/Pen%20module.png)
+   #### 3.Pen Lifting
+   Reck and Pinion mechanism helps us to lift the whole bot by providing perpendicular force on the wall while the pen module is in the process of rotation.
+ 
+  ![](Image%20and%20Videos/Mechanical%20Design%20Images/Pen%20lifting.png)
+## Electronics Aspect of the Design
+ In this project, OpenCV is used for image processing and the coordinates are stored in the form of a 30-30 array. Which are sent to arduino through serial transfer library. Then every coordinate is transformed into the length of two strings, which is again converted into number of steps of individual stepper motors.
```
Comment (theobscuredev) [inline README.md:70]: Elaborate the Electronics Aspect. Maybe include an image depicting the connections between Arduino and the necessary sensors using TinkerCAD. For eg: https://github.com/theobscuredev/Rube-Goldberg-Machine/blob/master/Images%20and%20Videos/Images/Line_Follower_Without_LSA.jpg

File: README.md [hunk 8]
```diff
@@ -1,2 +1,66 @@
...
+
+## Applications
+The wall art bot is planned to draw or sketch a picture which the user has in his laptop or phone.
```
Comment (theobscuredev) [inline README.md]: Elaborate the Applications.

File: README.md [hunk 9]
```diff
@@ -1,2 +1,66 @@
...
+1. [Aradhya Saxena](https://github.com/aries2001)
+2. [Bhunesh Gepal](https://github.com/Bhunesh22)
+3. [Kodumuru Firdose Kouser](https://github.com/fidosekouser1902)
+4. [Shikar Gupta](https://github.com/shikhar2624)
+5. [Trishit Mondal](https://github.com/trishit-byte)
+
+
+#### Mentors
+1. [Naman Agarwalla]
+2. [Parul Chaudhary]
```
Comment (theobscuredev) [inline README.md]: Hyperlink the GithubID of your mentors.

File: Image and Videos/Mechanical Design Images/1 [hunk 10]
```diff
@@ -0,0 +1 @@
+
```
Comment (theobscuredev) [inline Image and Videos/Mechanical Design Images/1]: Content Missing?

File: Image Processing/file [hunk 11]
```diff
@@ -0,0 +1 @@
+
```
Comment (theobscuredev) [inline Image Processing/file]: Missing Content? Either Delete this file or update.

File: Simulation/file [hunk 12]
```diff
@@ -0,0 +1 @@
+
```
Comment (theobscuredev) [inline Simulation/file]: Missing content? Either delete or update this file

File: Arduino/final_wall_art_2.ino [hunk 13]
```diff
@@ -0,0 +1,104 @@
...
+#define motorPin7  5    // IN3 on the ULN2003 driver 2
+#define motorPin8  6    // IN4 on the ULN2003 driver 2
+int stepperSpeed = 1000; //speed of the stepper (steps per second)
+float steps1 = 0; // keep track of the step count for motor 1
+float steps2=0;
+float target1=0;
+float target2=0;
+boolean turn1 = true;
+float n=0;
+// Initialize with pin sequence IN1-IN3-IN2-IN4 for using the AccelStepper with 28BYJ-48
```
Comment (theobscuredev) [inline Arduino/final_wall_art_2.ino]: Include line breaks and work on code aesthetics. Remove unnecessary and blank lines etc.

**Review by theobscuredev — CHANGES_REQUESTED** [review]
@aries2001 Changes are required!

1. The 'src' folder is missing!! Include the Arduino & Image Processing code in the src folder. Follow proper file structure as mentioned in the documentation guidelines.
2. It is highly suggested you follow a uniform naming convention for all files in each of the directories.. Either resort to camelCase (For eg: AlanWalkerLogo.mp4) or underscores (Alan_Walker_Logo.mp4) or hyphens (Alan-Walker-Logo.mp4), but make sure it is uniform.
2. Make the suggested changes in the README file. Particularly,
       a. Look out for line breaks and spaces. Focus on the aesthetics.
       b. Center align all the images and justify the text using the suggested tags. Tabulate the 
           cost structure and resize few images.
       c. Refer the guidelines for the section order. Cost Structure should be included after 
           Mechanical and Electronics Aspects.
       d. Hyperlinking
3. Prepare a poster (in jpeg format) containing the proper content & logos. Refer the guidelines for the same.
4. Rename the folders as per the guidelines. For eg: Poster and Report, Images and Videos etc. , they should be named as mentioned in the guidelines.
5. Create separate 'Images' & 'Videos' folder inside the 'Images and Videos' directory and add the corresponding files in their respective folders. Remove the workflow image from 'Poster and Report' and include it in the 'Images' folder.
6. Review the entire codebase and focus on line breaks, indentation & spaces. Work on the aesthetics. Remove unnecessary lines and add comments wherever necessary.
7. Rename all the files. Avoid naming them as 'final_2.py' etc. Naming should be self-explanatory and indication of the content included in the same.

**Comment by theobscuredev** [comment]
@shikhar2624 @trishit-byte Lemme know once you've made the final changes. Make sure to tag me (@) in this thread.

File: README.md [hunk 14]
```diff
@@ -1,2 +1,116 @@
 # Wall-Art-Bot
 Open Projects 2021
```
Comment (theobscuredev) [inline README.md]: Apply header using '## Open Projects 2021'

File: README.md [hunk 15]
```diff
@@ -1,2 +1,116 @@
...
+#### Mentors
+1. [Naman Agarwalla](https://github.com/naman99-agar)
+2. [Parul Chaudhary](https://github.com/Parul253)
+
+## References
+* http://42bots.com/tutorials/28byj-48-stepper-motor-with-uln2003-driver-and-arduino-uno/
+* https://github.com/PowerBroker2/pySerialTransfer
+* https://www.youtube.com/watch?v=T0jwdrgVBBc&t=72s&ab_channel=MERTArduino%26Tech
+* https://www.youtube.com/watch?v=FbR9Xr0TVdY&ab_channel=ProgrammingKnowledge
+* https://www.hackster.io/fredrikstridsman/stringent-the-15-wall-plotter-d965ca#overview
```
Comment (theobscuredev) [inline README.md]: Hyperlink the References properly. No need to link the URLs again. Refer this README: https://raw.githubusercontent.com/marsiitr/Rube-Goldberg-Machine/master/README.md

**Review by theobscuredev — CHANGES_REQUESTED** [review]
@shikhar2624 Insert 's' while naming the 'Images and Videos' folder and update readme. Just these final changes and we're good to go

**Comment by theobscuredev** [comment]
@shikhar2624 & team,
Make the following final changes:

1. Rectify the naming of the folder: ~~Image and Videos~~ -> Images and Videos
2. Try to elaborate the control part in a few statements after the diagram.

Make these changes and then we're good to go!

**Comment by theobscuredev** [comment]
Nice work!
@SanjeevKrishnan I suggest for merge

**Review by SanjeevKrishnan — CHANGES_REQUESTED** [review]
README.md looks nice!
Few changes are expected
1. Folder "Wall-Art-Bot/Image and Videos/Electronics Mechanism" is not making any sense. Shift the contents of "Electronics Mechanism" into "Wall-Art-Bot/Image and Videos/Images"
2. Folder "Wall-Art-Bot/Image and Videos/Mechanical Design Images" has to be present under "Wall-Art-Bot/Mechanical Design" folder, not under "Wall-Art-Bot/Images and Videos"
3. Rename "wall art bot.pdf" as "Report.pdf"
4. Create a subfolder "Matlab" under the folder "Wall-Art-Bot/Simulation" and shift the contents of "Wall-Art-Bot/Simulation" into "Wall-Art-Bot/Simulation/Matlab". Add "Wall-Art-Bot/Simulation/README.md" and elaborate on simulation.
5. Create a subsection "Simulation" in "Wall-Art-Bot/README.md" and give a summary and reference of "Wall-Art-Bot/Simulation/README.md" in that.
For points 4 and 5, refer to [RoverSim Documentation](https://github.com/marsiitr/RoverSim).
6. Cost Structure must include cost for manufacturing the bot. In your case, 3D printing cost has to be included.

**Review by SanjeevKrishnan — APPROVED** [review]
Great efforts team! Impressive work
PR looks good. Merging the PR

**Pull request merged**